netmiko/garderos/garderos_ssh.py

```diff
@@ -0,0 +1,488 @@
+
+        Automatically exits/enters configuration mode.
+
+        :param config_commands: Multiple configuration commands to be sent to the device
+
+        :param exit_config_mode: Determines whether or not to exit config mode after complete
+
+        :param delay_factor: Deprecated in Netmiko 4.x. Will be eliminated in Netmiko 5.
+
+        :param max_loops: Deprecated in Netmiko 4.x. Will be eliminated in Netmiko 5.
+
+        :param strip_prompt: Determines whether or not to strip the prompt
+
+        :param strip_command: Determines whether or not to strip the command
+
+        :param read_timeout: Absolute timer to send to read_channel_timing. Also adjusts
+        read_timeout in read_until_pattern calls.
+
+        :param config_mode_command: The command to enter into config mode
+
+        :param cmd_verify: Whether or not to verify command echo for each command in config_set
+
+        :param enter_config_mode: Do you enter config mode before sending config commands
+
+        :param error_pattern: Regular expression pattern to detect config errors in the
+        output.
+
+        :param terminator: Regular expression pattern to use as an alternate terminator in certain
+        situations.
+
+        :param bypass_commands: Regular expression pattern indicating configuration commands
+        where cmd_verify is automatically disabled.
+        """
+        # The result of all commands will be collected to config_results
+        config_results = ""
+        # Set delay_factor to given value
+        if delay_factor is None:
+            delay_factor = self.select_delay_factor(0)
+        else:
+            delay_factor = self.select_delay_factor(delay_factor)
+        # Verify if config_commands is an array
+        if config_commands is None:
+            return config_results
+        elif isinstance(config_commands, str):
+            config_commands = (config_commands,)
+        if not hasattr(config_commands, "__iter__"):
+            raise ValueError("Invalid argument passed into send_config_set")
+        # Go to config mode. Use given config_mode_command if necessary.
+        if enter_config_mode:
+            if config_mode_command:
+                config_results += self.config_mode(config_mode_command)
+            else:
+                config_results += self.config_mode()
+        # Send all commands to the router and verify their successful execution
+        for command in config_commands:
+            # Verification is done in send_config_command() function
+            # Will raise error on execution failure
+            result = self.send_config_command(command)
+            config_results = f"{command}\n{result}\n"
+        # Exit config mode if needed
+        if exit_config_mode:
+            self.exit_config_mode()
+        # Return all results
+        # Will only be executed if no error occured
+        return config_results
+
+    def ssh_connect(
```

Comment:
What is the purpose this? Probably should be outside of Netmiko.

Reply:
The intention was to have a nested SSH connection inside the netmiko session without the need of using JumpHosts or Redispatch. I thought this could be useful to others, too.
I suggest to remove this from the driver and move it to the user code.

Reply:
Removed in updated pull-request